test/ports_test.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2012-2018 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/nats-io/gnatsd/server"
+)
+
+// waits until a calculated list of listeners is resolved or a timeout
+func waitForFile(path string, dur time.Duration) ([]byte, error) {
+	end := time.Now().Add(dur)
+	for time.Now().Before(end) {
+		if _, err := os.Stat(path); os.IsNotExist(err) {
+			time.Sleep(25 * time.Millisecond)
+			continue
+		} else {
+			return ioutil.ReadFile(path)
+		}
+	}
+	return nil, errors.New("Timeout")
+}
+
+func TestPortsFile(t *testing.T) {
+	opts := DefaultTestOptions
+	opts.PortsFileDir = os.TempDir()
+	opts.HTTPPort = -1
+	opts.ProfPort = -1
+	opts.Cluster.Port = -1
+
+	s := RunServer(&opts)
+	if !s.ReadyForListeners(nil, 5*time.Second) {
```

Comment:
See, not sure there is a use case for passing options here. Anyway, you want the options that the server has, not the one that you would provide that could be different than what the server was started with.